db/python/tables/cohort.py

```diff
@@ -185,14 +192,36 @@ async def create_cohort(
             VALUES (:cohort_id, :sequencing_group_id, :audit_log_id)
             """
 
-            for sg in sequencing_group_ids:
-                await self.connection.execute(
-                    _query,
+            await self.connection.execute_many(
+                _query,
+                [
                     {
                         'cohort_id': cohort_id,
                         'sequencing_group_id': sg,
                         'audit_log_id': audit_log_id,
-                    },
-                )
+                    }
+                    for sg in sequencing_group_ids
+                ],
+            )
+
+            return NewCohortInternal(
+                dry_run=False,
+                cohort_id=cohort_id,
+                sequencing_group_ids=sequencing_group_ids,
+            )
+
+    async def get_cohort_by_id(self, cohort_id: int) -> CohortInternal:
```

Comment:
FWIW, the `get_by_id` could be implemented through the query method, with a `CohortFilter(id=GenericFilter(eq=cohort_id))`. Not asking for this as it's usually simpler code to do this separately. Just wanted to flag it as possible.